src/plonk/lookup/prover.rs

```diff
@@ -627,18 +664,28 @@ fn permute_expression_pair<C: CurveAffine>(
     }
     assert!(repeated_input_rows.is_empty());
 
-    let mut permuted_table_expression = domain.empty_lagrange();
-    parallelize(
-        &mut permuted_table_expression,
-        |permuted_table_expression, start| {
-            for (permuted_table_value, permuted_table_coeff) in permuted_table_expression
-                .iter_mut()
-                .zip(permuted_table_coeffs[start..].iter())
-            {
-                *permuted_table_value += permuted_table_coeff;
+    permuted_input_expression.extend((0..(blinding_factors + 1)).map(|_| C::Scalar::rand()));
+    permuted_table_coeffs.extend((0..(blinding_factors + 1)).map(|_| C::Scalar::rand()));
+    assert_eq!(permuted_input_expression.len(), params.n as usize);
+    assert_eq!(permuted_table_coeffs.len(), params.n as usize);
+
+    #[cfg(feature = "sanity-checks")]
+    {
+        let mut last = C::Scalar::rand();
```

Comment:
This is confusing.
```suggestion
        let mut last = None;
```

Reply:
Sorry :) wrote it while debugging and thought I'd delete this anyway, but it's not really a bad sanity check to leave in.